src/wp-includes/blocks/search.php

```diff
@@ -180,6 +206,25 @@ function register_block_core_search() {
 }
 add_action( 'init', 'register_block_core_search' );
 
+/**
+ * Ensure that the view script has the `wp-interactivity` dependency.
+ *
+ * @since 6.4.0
+ *
+ * @global WP_Scripts $wp_scripts
+ */
+function block_core_search_ensure_interactivity_dependency() {
+	global $wp_scripts;
+	if (
+		isset( $wp_scripts->registered['wp-block-search-view'] ) &&
+		! in_array( 'wp-interactivity', $wp_scripts->registered['wp-block-search-view']->deps, true )
+	) {
+		$wp_scripts->registered['wp-block-search-view']->deps[] = 'wp-interactivity';
+	}
+}
+
+add_action( 'wp_print_scripts', 'block_core_search_ensure_interactivity_dependency' );
```

Comment:
See above, it would be better to rely on the search block's `supports.interactivity` field to centrally set this dependency.